20_aws_networking/aws_vpc.tf

```diff
@@ -0,0 +1,76 @@
+resource "aws_vpc" "main" {
+  cidr_block           = var.vpc_settings.cidr
+  enable_dns_support   = var.vpc_settings.enable_dns_support
+  enable_dns_hostnames = var.vpc_settings.enable_dns_hostnames
+
+  tags = {
+    Name = format(local.resource_name, var.vpc_settings.name)
+  }
+}
+
+resource "aws_subnet" "public" {
+  for_each          = var.subnet_settings.public
+  vpc_id            = aws_vpc.main.id
+  cidr_block        = local.public_subnet_cidrs[each.key]
+  availability_zone = element(data.aws_availability_zones.available.names, index(keys(var.subnet_settings.public), each.key))
+
+  map_public_ip_on_launch = each.value.map_public_ip_on_launch
+
+  tags = {
+    Name = local.public_subnet_names[each.key]
+  }
+}
+
+resource "aws_subnet" "private" {
+  for_each          = var.subnet_settings.private
+  vpc_id            = aws_vpc.main.id
+  cidr_block        = local.private_subnet_cidrs[each.key]
+  availability_zone = element(data.aws_availability_zones.available.names, index(keys(var.subnet_settings.private), each.key))
+
+  tags = {
+    Name = local.private_subnet_names[each.key]
+  }
+}
+
+resource "aws_internet_gateway" "main_igw" {
+  vpc_id = aws_vpc.main.id
+
+  tags = {
+    Name = format(local.resource_name, var.igw_settings.name)
+  }
+}
+
+resource "aws_route_table" "public" {
+  vpc_id = aws_vpc.main.id
+
+  route {
+    cidr_block = var.public_route_table_settings.routes.internet.cidr_block
+    gateway_id = aws_internet_gateway.main_igw.id
+  }
+
+  tags = {
+    Name = format(local.resource_name, var.public_route_table_settings.name)
+  }
+}
+
+resource "aws_route_table" "private" {
+  vpc_id = aws_vpc.main.id
+
+  #   TODO Add dynamic assigning of NAT instance to the private route table on ASG events
+
```

Comment:
Should be added in nat instance user-data. Ignore some resources in terraform to avoid drifts.